blackjax/mcmc/elliptical_slice.py

```diff
@@ -0,0 +1,166 @@
+"""Public API for the Elliptical Slice sampling Kernel"""
+from typing import Callable, NamedTuple, Tuple
+
+import jax
+import jax.numpy as jnp
+
+from blackjax.types import Array, PRNGKey, PyTree
+
+__all__ = ["EllipSliceState", "EllipSliceInfo", "init", "kernel"]
+
+
+class EllipSliceState(NamedTuple):
+    """State of the Elliptical Slice sampling algorithm.
+
+    position
+        Current position of the chain.
+    loglikelihood
+        Current value of the log likelihood only.
+    """
+
+    position: PyTree
+    loglikelihood: PyTree
+
+
+class EllipSliceInfo(NamedTuple):
+    r"""Additional information on the Elliptical Slice sampling chain.
+
+    This additional information can be used for debugging or computing
+    diagnostics.
+
+    momentum
+        The latent momentum variable returned at the end of the transition.
+    theta
+        A value between [-2\pi, 2\pi] identifying points in the ellipsis drawn
+        from the positon and mommentum variables. This value indicates the theta
+        value of the accepted proposal.
+    subiter
+        Number of sub iterations needed to accept a proposal. The more subiterations
+        needed the less efficient the algorithm will be, and the more dependent the
+        new value is likely to be to the previous value.
+    """
+
+    momentum: PyTree
+    theta: float
+    subiter: int
+
+
+def init(position: PyTree, loglikelihood_fn: Callable):
+    loglikelihood = loglikelihood_fn(position)
+    return EllipSliceState(position, loglikelihood)
+
+
+def kernel(cov_matrix: Array, mean: Array):
+    """Build an Elliptical Slice sampling kernel.
+
+    Parameters
+    ----------
+    cov_matrix
+        The value of the covariance matrix of the gaussian prior distribution from the posterior we wish to sample.
+
+    Returns
+    -------
+    A kernel that takes a rng_key and a Pytree that contains the current state
+    of the chain and that returns a new state of the chain along with
+    information about the transition.
+
+    """
+
+    cov_matrix_sqrt = jax.lax.linalg.cholesky(cov_matrix)
```

Comment:
Is this true in the limiting case where `cov_matrix` is a scalar? Check `metrics.py`